google/cloud/internal/grpc_opentelemetry.cc

```diff
@@ -145,6 +145,13 @@ void ExtractAttributes(grpc::ClientContext& context,
   }
 }
 
+bool IsInitialMetadataReady(StatusCode code) {
+  // We assume that if the status code is one of the following
+  // the initial metadata *might* not be ready.
+  // See go/cloud-cxx:grpc-workaround for details.
+  return (code != StatusCode::kCancelled);
```

Comment:
Let's start with what we expect the answer to be. `kCancelled`, `kUnavailable`, `kDeadlineExceeded`.